src/test/java/com/atlassian/bitbucket/jenkins/internal/fixture/BitbucketJenkinsRule.java

```diff
@@ -60,12 +60,19 @@ public void before() throws Throwable {
                 "", BITBUCKET_ADMIN_USERNAME, ADMIN_PERSONAL_TOKEN.get().getSecret());
         addCredentials(readCredentials);
 
+        createRepoFork();
```

Comment:
I think a possible danger here is if some error occurs during the testing, and after isn't called or fails, then every other func test relying on the BBJenkinsRule will also fail, because this call's expect clause will fail (repo already exists). 
We could remove that expect but that makes diagnosing tests that rely on that fork a bit more frustrating. We could also attempt a delete before the create but that feels inelegant and adds overhead.
Open to suggestions on how to make this flexible- this feels like a potential source of flakiness.

Reply:
Why don't you use a random project key so that there's no chance of that happening